docs/site/Controllers.md

```diff
@@ -269,3 +269,34 @@ export class HelloController {
   }
 }
 ```
+
+## Creating Controllers at Runtime
+
+A controller can be created for a model at runtime using the
+`defineCrudRestController` helper function from the `@loopback/rest-crud`
+package. It accepts a Model class and a `CrudRestControllerOptions` object.
+Dependency injection for the controller has to be configured by applying the
+`inject` decorator manually as shown in the example below.
+
+```ts
+const basePath = '/' + bookDef.name;
+const BookController = defineCrudRestController(BookModel, {basePath});
+inject(repoBinding.key)(BookController, undefined, 0);
+```
+
+The controller is then attached to the app by calling the `app.controller()`
+method.
+
+```ts
+app.controller(BookController);
+```
+
+The new CRUD REST endpoints for the model will be available on the app now.
+
+If you want a customized controller, you can create a copy of
+`defineCrudRestController`'s
+[implementation](https://github.com/strongloop/loopback-next/blob/00917f5a06ea8a51e1f452f228a6b0b7314809be/packages/rest-crud/src/crud-rest.controller.ts#L129-L269)
+and modify it according to your requirements.
+
+For details about `defineCrudRestController` and `CrudRestControllerOptions`,
+refer to the [@loopback/rest-crud API documentation](./apidocs/rest-crud.html).
```

Comment:
Maybe we should mention Class factory to allow parameterized decorations per https://loopback.io/doc/en/lb4/core-tutorial-part10.html (Class factory to allow parameterized decorations).

Reply:
+1

Reply:
Not really convinced. If I am understanding correctly, the content at the link describes a general programming pattern, not something specific to Controllers. If that's the case, we should avoid adding non-specific content to avoid noise.

Reply:
The class factory is part of the programming model pattern and it's being asked by our users. It gives our users more control. The change is applying to `Controllers.md` and I don't think that's noise.

Reply:
> The change is applying to `Controllers.md`

Got the context, I thought it was about runtime controllers.

Added **Class factory to allow parameterized decorations** at an appropriate location.